webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/design/SiteDesignActionHandler.java

```diff
@@ -0,0 +1,660 @@
+  }
+
+  private String parseCodeSupprImage(WebSiteSessionController scc, String code,
+      HttpServletRequest request, SettingBundle settings, String currentPath) {
+    String avant;
+    String apres;
+    int index;
+    String finChemin;
+    String image =
+        "<IMG border=0 src=\"http://" + getMachine(request) + "/" + settings.getString("Context") +
+            "/" + scc.getComponentId() + "/";
+    int longueurImage = 19 +
+        ("http://" + getMachine(request) + "/" + settings.getString("Context") + "/" +
+            scc.getComponentId() + "/").length();
+    index = code.indexOf(image);
+    if (index == -1) {
+      return code;
+    } else {
+      avant = code.substring(0, index + 19);
+      finChemin = code.substring(index + longueurImage);
+
+      int indexGuillemet = finChemin.indexOf("\"");
+      String absolute = finChemin.substring(0, indexGuillemet);
+
+      apres = finChemin.substring(indexGuillemet);
+      int indexSlash = absolute.lastIndexOf("/");
+      String fichier = absolute.substring(indexSlash + 1);
+
+      String deb = absolute.substring(0, indexSlash);
+      List<String> tab = buildTab(deb + "/");
+
+      /* id/rep1 */
+      String cheminContexte = finNode(scc, currentPath);
+      List<String> tabContexte = buildTab(cheminContexte + "/");
+      List<String> tabCommun = sortCommun(tabContexte, tab);
+      String reste = sortReste(tab, tabCommun);
+      int nbPas = tabContexte.size() - tabCommun.size();
+      String relatif = "";
+      int i = 0;
+      while (i < nbPas) {
+        relatif += "../";
+        i++;
+      }
+
+      if (reste.equals("")) {
+        relatif += fichier;
+      } else {
+        relatif += reste + "/" + fichier;
+      }
+      apres = relatif + apres;
+      return (avant + parseCodeSupprImage(scc, apres, request, settings, currentPath));
+    }
+  }
+
+  private String parseCodeSupprHref(WebSiteSessionController scc, String code,
+      SettingBundle settings, String currentPath) {
+    String avant;
+    String apres;
+    int index;
+    String href = "<A href=\""; /* longueur de chaine = 9 */
+
+    String finChemin;
+    String fichier;
+    String deb;
+    String theReturn = "";
+
+    index = code.indexOf(href);
+    if (index == -1) {
+      theReturn = code;
+    } else {
+
+      avant = code.substring(0, index + 9);
+
+      apres = code.substring(index + 9);
+
+      if (apres.substring(0, 7).equals("http://")) { /* lien externe */
+
+        theReturn = avant + parseCodeSupprHref(scc, apres, settings, currentPath);
+      } else if (apres.substring(0, 6).equals("ftp://")) { /* lien externe */
+
+        theReturn = avant + parseCodeSupprHref(scc, apres, settings, currentPath);
+      } else if (apres.substring(0, 3).equals("rr:")) { /* deja en relatif */
+
+        apres = apres.substring(3);
+
+        theReturn = avant + parseCodeSupprHref(scc, apres, settings, currentPath);
+      } else if (apres.substring(0, 3).equals("aa:")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "aa:" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1132)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)